docs/test_marking.md

```diff
@@ -0,0 +1,183 @@
+# Source code elements related to pytest
+## Contents
+- [Test case description](#Test_case_description)
+  - [Test marking](#Test_marking)
+    - [Requirements](#Requirements)
+    - [Type](#Type)
+    - [Priority](#Priority)
+    - [Component](#Component)
+    - [Skip](#Skip)
+    - [Bug(s)](#Bug)
+    - [Parameters](#Parameters)
+    - [Fixtures](#Fixtures)
+  - [Test naming convention](#TestNaming)
+    - [Test name](#TestName)
+  - [Test documentation](#TestDoc)
+    - [Docstring](#Docstring)
+    - [Steps in test body](#Step)
+          
+<a id="Test_case_description"></a>
+## Test case description
+<a id="Test_marking"></a>
+### Test marking
+<a id="Requirements"></a>
+#### Requirements 
+Fully defined in GitHub issues, e.g.
+```
+    @pytest.mark.reqids(Requirements.DATUM_244, Requirements.DATUM_333)
+```
+Requirements constants need to be added to datumaro/tests/constants/requirements.py for example:
+```
+# [DATUMARO] Add SNYK scan integration
+DATUM_244 = "DATUM-244 Add Snyk integration"
+```  
+Recommended notation for requirements 
+
+```python
+@pytest.mark.requids(Requirements.DATUM_123)
+``` 
+where DATUM is a keyword for datumaro indication, and number is a datumaro github issue number.
+
+<a id="Type"></a>
+#### Type
+Test types: gui_smoke, gui_regression, manual, gui_other, gui_long, api, component, unit e.g.
+```python
+    @pytest.mark.component
+    @pytest.mark.unit
+```
+
+<a id="Priority"></a>
+#### Priority 
+Test priorities: low, medium, high, e.g.,
+```python
+     @pytest.mark.priority_low
+     @pytest.mark.priority_medium
+     @pytest.mark.priority_high
+```
+<a id="Component"></a>
+#### Component
+
+Component marking used for indication of different system components e.g.
+```python
+    @pytest.mark.components(DatumaroComponent.Datumaro)
+```
+<a id="Skip"></a>
+#### Skip 
+
+For marking tests, which should be skipped for example for not yet tests, e.g.,
+```python
+   @pytest.mark.skip(reason=SkipMessages.NOT_IMPLEMENTED)
+```
+<a id="Bug"></a>
+#### Bug(s):
+
+In case of test failure, bug should be entered into github issues, and test can be marked e.g.
+```python
+    @pytest.mark.bugs("DATUM-219 - Return format is not uniform")
+```
+<a id="Parameters"></a>
+#### Parameters: 
+
+Parameters are used for running the same test with different parameters e.g. 
+```python
+@pytest.mark.parametrize("numpy_array, batch_size", [  
+        (np.zeros([2]), 0),  
+        (np.zeros([2]), 1),
+        (np.zeros([2]), 2),
+        (np.zeros([2]), 5),
+        (np.zeros([5]), 2),
+    ])
+```
+(be aware that test parametrization is supported in pytest, and is not supported in unittests) 
```

Comment:
I don't see a reason to support both frameworks. We need to select one and support it, otherwise it is going to become a mess with no value.

Reply:
I would like to recommend to switch to pytest. A few advantages were mentioned in summary: - 
- running tests parallelly by multiple processes
- run the same test a few times with different parameters
- marking tests with predefined marks, and use these marks to run only selected group of tests
- generating html reports if needed with installed pytest-html library.